### DISCUSSION TITLE
How could I express optional tuple element?

### DISCUSSION THREAD

**Discussion by ellemedit** [open-circle/valibot#311] [General]
Following is considered as valid (not thrown):

```js
parse(tuple([string(), optional(string())]), ['test', undefined]);
```

Following is considered as invalid (thrown):

```js
parse(tuple([string(), optional(string())]), ['test']);
```

Is this intended behavior? I thought I could skip the rest optional tuple element.

**Comment by fabian-hiller** [comment]
`optional` behave this way only for `object`. A `tuple` requires a fixed min length. Feel free to create an issue. I will then investigate this case to see if we want to change or improve this behavior. For now, maybe you can use the `rest` argument as a workaround:

```ts
parse(tuple([string()], string()), ['test']);
```

**Reply by ellemedit** [reply]
Thank you for suggesting workaround!

I'll discuss with my colleagues and twitters. I'll open an issue if needed.

**Reply by ellemedit** [reply]
tuple rest parameters considered as a optional parameters?

**Reply by fabian-hiller** [reply]
Yes. But it allows an infinite amount of it at the end of the tuple. So `['test', 'test', 'test', 'test']` is considered a valid input.
